File: solution/1508. Range Sum of Sorted Subarray Sums/1508_01.cpp
class Solution {
public:
    int rangeSum(vector<int>& nums, int n, int left, int right) {
        const int MOD = 1e9 + 7;
        vector<int> A;
        for (int i = 0; i < n; ++i) {
            int sum = 0;
            for (int j = i; j < n; ++j) {
                sum += nums[j];
                A.push_back(sum);
            }
        }
        sort(A.begin(), A.end());
        long res = 0;
        for (int i = left - 1; i < right; ++i)
            res = (res + A[i]) % MOD;
        return res;
    }
};

File: solution/1508. Range Sum of Sorted Subarray Sums/1508_01.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        sums = []
        for i in range(n):
            s = 0
            for j in range(i, n):
                s += nums[j]
                sums.append(s)
        sums.sort()
        return sum(sums[left - 1:right]) % 1_000_000_007
    
